activejob/lib/active_job/logging.rb

```diff
@@ -87,12 +87,25 @@ def queue_name(event)
         def args_info(job)
           if job.arguments.any?
             ' with arguments: ' +
-              job.arguments.map { |arg| arg.try(:to_global_id).try(:to_s) || arg.inspect }.join(', ')
+              job.arguments.map { |arg| serialize_arg(arg).inspect }.join(', ')
           else
             ''
           end
         end
 
+        def serialize_arg(arg)
+          case arg
+          when Hash
+            arg.each_with_object({}) { |(key, value), hash| hash[key] = serialize_arg(value)}
```

Comment:
We could use `transform_values`:

```ruby
when Hash
  arg.transform_values { |value| format(value) }
when # ...
```

To get that, you need to require `active_support/core_ext/hash/transform_values`.